src/main/java/io/pinecone/PineconeDataPlaneClient.java

```diff
@@ -0,0 +1,222 @@
+                .setId(id)
+                .addAllValues(values);
+
+        if (sparseIndices != null && sparseValues != null && sparseIndices.size() != sparseValues.size()) {
+            throw new PineconeValidationException("Invalid Upsert Request. Please ensure that both sparse indices and values are of the same length.");
+        }
+
+        if (sparseIndices != null && !sparseIndices.isEmpty()) {
+            vectorBuilder.setSparseValues(SparseValues.newBuilder()
+                    .addAllIndices(convertUnsigned32IntToSigned32Int(sparseIndices))
+                    .addAllValues(sparseValues)
+                    .build());
+        }
+
+        if (metadata != null) {
+            vectorBuilder.setMetadata(metadata);
+        }
+
+        upsertRequest.addVectors(vectorBuilder.build());
+
+        if (namespace != null && !namespace.isEmpty()) {
+            upsertRequest.setNamespace(namespace);
+        }
+
+        return blockingStub.upsert(upsertRequest.build());
+    }
+
+    public QueryResponse query(String id,
+                               String namespace,
+                               int topK,
+                               Struct filter,
+                               boolean includeValues,
+                               boolean includeMetadata,
+                               List<Float> vectors,
+                               List<Long> sparseIndices,
+                               List<Float> sparseValues) {
+        QueryRequest.Builder queryRequest = QueryRequest.newBuilder();
+
+        if (id != null && !id.isEmpty() && vectors != null && !vectors.isEmpty()) {
+            throw new PineconeValidationException("Invalid Query Request. Both Vector ID and vectors cannot be present.");
+        }
+
+        if (id != null && !id.isEmpty()) {
+            queryRequest.setId(id);
+        }
+
+        if (namespace != null && !namespace.isEmpty()) {
+            queryRequest.setNamespace(namespace);
+        }
+
+        queryRequest.setTopK(topK)
+                .setIncludeValues(includeValues)
+                .setIncludeMetadata(includeMetadata);
+
+        if (filter != null) {
+            queryRequest.setFilter(filter);
+        }
+
+        if (vectors != null && !vectors.isEmpty()) {
+            queryRequest.addAllVector(vectors);
+        }
+
+        if (sparseIndices != null && sparseValues != null) {
+            if (sparseIndices.size() != sparseValues.size()) {
+                throw new PineconeValidationException("Invalid Upsert Request. Please ensure that both sparse indices and values are of the same length.");
+            }
+
+            queryRequest.setSparseVector(SparseValues.newBuilder()
+                    .addAllIndices(convertUnsigned32IntToSigned32Int(sparseIndices))
+                    .addAllValues(sparseValues)
+                    .build());
+        }
+
+        return blockingStub.query(queryRequest.build());
+    }
+
+    public FetchResponse fetch(List<String> ids,
```

Comment:
I would add an overload method for `fetch(List<String> ids)` without namespace so that people who aren't using namespaces will not have to pass the default namespace value `""`.

Reply:
Done!